t/t9000-system.t

```diff
@@ -0,0 +1,53 @@
+#!/bin/sh
+#
+
+test_description='Run tests against a system instance of Flux'
+
+. `dirname $0`/sharness.sh
+
+owner=$(flux getattr security.owner)
+if test -n "$owner" -a "$owner" != "$(id -u)"; then
+    	test_set_prereq SYSTEM
+fi
```

Comment:
Looks great!

It might also be good to check for  `FLUX_TEST_INSTALLED_PATH` before setting the SYSTEM prereq, so we don't end up testing a mixture of flux versions here.

Reply:
Good point!

Note that in the `system` CI image, the version of Flux installed in the fluxorama image should be exactly the same as the version in the build directory. However, testing from builddir is not necessarily a good test that everything works properly when installed (and is installed in the right place), so probably best to set up the environment to use installed paths.

Reply:
For `t9000-system.t` I think we always want to test the installed path so how about this?
```diff
diff --git a/t/t9000-system.t b/t/t9000-system.t
index 4ee53083f..0b820de04 100755
--- a/t/t9000-system.t
+++ b/t/t9000-system.t
@@ -3,6 +3,7 @@
 
 test_description='Run tests against a system instance of Flux'
 
+FLUX_TEST_INSTALLED_PATH=${FLUX_TEST_INSTALLED_PATH:-/usr/bin}
 . `dirname $0`/sharness.sh
 
 owner=$(flux getattr security.owner)

```

I "tested" that this runs the installed `flux` by running the test after `make clean` completed at the top level:

```
 grondo@fluxorama:/usr/src/t$ ../src/cmd/flux
bash: ../src/cmd/flux: No such file or directory
 grondo@fluxorama:/usr/src/t$ ./t9000-system.t  -d -v
sharness: loading extensions from /usr/src/t/sharness.d/01-setup.sh
sharness: loading extensions from /usr/src/t/sharness.d/flux-sharness.sh
expecting success: 
	jobid=$(flux mini submit id -u) &&
	result=$(flux job attach $jobid) &&
	test_debug "echo Job ran with userid $result" &&
	test $result -eq $(id -u) &&
	test $(flux getattr security.owner) -ne $result

Job ran with userid 1000
ok 1 - 0001-basic.t: system instance runs job as current uid

expecting success: 
	test $(flux jobs -no {userid} $jobid) -eq $(id -u)

ok 2 - 0001-basic.t: flux jobs lists job with correct userid

# passed all 2 test(s)
1..2

```

Reply:
I was just thinking about running `make check` on a system that happens to have a different versin flux system instance running.

Reply:
Ah, I see, and we wouldn't want to run this test which might not be appropriate for the system installed version.
Hm, I'm not certain how to work around that one. Maybe a separate environment variable that is required for force the test to run, kind of like we have for the valgrind test?

Reply:
Could we just skip the test if FLUX_TEST_INSTALLED is unset?

Reply:
I could be confused. Is that set by default for the system instance testing?
AFAIK, that is set by the `test-install` builder onlly.

Maybe it should be set for the system instance testing, though we'd have to figure out if coverage works for installed binaries and libraries.

Reply:
Yeah I was thinking just set that for the system instance testing.  Sorry, I wasn't thinking about the coverage issue.  Your proposal of an "opt-in" variable is ok with me!

Reply:
Yep, agreed!